### PR TITLE
Create a mongo client factory

### PR DESCRIPTION
**Pull request by sfriedlizuehlke**
Add the following configuration options for the mongo client:
- `username`
- `password`
- `tls`

### CODE REVIEW

File: polytope_server/common/mongo_client_factory.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import pymongo
+import urllib.parse
+import typing
```
Comment (cfkanesan) [inline polytope_server/common/mongo_client_factory.py]: nit: imports should be grouped into std, 3rd party, 1st party

**Review by cfkanesan — APPROVED** [review]

File: polytope_server/common/mongo_client_factory.py (hunk 1 above)
Comment (sfriedlizuehlke) [reply polytope_server/common/mongo_client_factory.py]: [2ace679](https://github.com/MeteoSwiss-APN/polytope-server/pull/8/commits/2ace679f3100330308176fc136fc9707ecd81086)

**Review by cfkanesan — APPROVED** [review]

File: polytope_server/common/authentication/mongodb_authentication.py [hunk 2]
```diff
@@ -37,9 +36,12 @@ def __init__(self, name, realm, config):
         host = config.get("host", "localhost")
         port = config.get("port", "27017")
         collection = config.get("collection", "users")
+        username = config.get("username")
+        password = config.get("password")
+        tls = config.get("tls", False) == True
```
Comment (victoria-cherkas) [inline polytope_server/common/authentication/mongodb_authentication.py:41]: `class MongoAuthentication `now appears in two places, is that the aim?

Comment (victoria-cherkas) [reply polytope_server/common/authentication/mongodb_authentication.py:41]: actually name is slightly different (`class MongoDBAuthorization`), but still not sure why both are needed?

Comment (cfkanesan) [reply polytope_server/common/authentication/mongodb_authentication.py:41]: Usually, authentication checks that the user is who he/she claims to be. Authorization checks that the user actually has access to the resource that the user is trying to access. For consistency, DB should be in both names. However, I think Stephan's goal is to make as few changes as possible.

**Review by milosbeliczuhlke — APPROVED** [review]

File: polytope_server/common/authentication/mongodb_authentication.py (hunk 2 above)
Comment (sfriedlizuehlke) [reply polytope_server/common/authentication/mongodb_authentication.py:41]: Every component (request store, auth, ...) has its own mongo client. For now the the goal was to just add a way of providing credentials to the mongo client.

Comment (victoria-cherkas) [reply polytope_server/common/authentication/mongodb_authentication.py:41]: Ok thanks for the explanation, I missed the difference

**Review by victoria-cherkas — APPROVED** [review]

**Pull request merged**